src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -972,6 +973,400 @@ static void Main()
+                Diagnostic(ErrorCode.ERR_AmbigCall, "F1").WithArguments("Program.F1(System.Span<int>, int[])", "Program.F1(int[], System.ReadOnlySpan<int>)").WithLocation(10, 9),
+                // (11,9): error CS0121: The call is ambiguous between the following methods or properties: 'Program.F2(string, string[])' and 'Program.F2(object, Span<string>)'
+                //         F2("3", ["4"]);
+                Diagnostic(ErrorCode.ERR_AmbigCall, "F2").WithArguments("Program.F2(string, string[])", "Program.F2(object, System.Span<string>)").WithLocation(11, 9));
+        }
+
+        // Two ref struct collection types, with an implicit conversion from one to the other.
+        [Fact]
+        public void BetterConversionFromExpression_06()
+        {
+            string source = """
+                using System;
+                using System.Collections.Generic;
+                using System.Runtime.CompilerServices;
+                [CollectionBuilder(typeof(MyCollectionBuilder), nameof(MyCollectionBuilder.Create1))]
+                ref struct MyCollection1<T>
+                {
+                    private readonly List<T> _list;
+                    public MyCollection1(List<T> list) { _list = list; }
+                    public IEnumerator<T> GetEnumerator() => _list.GetEnumerator();
+                    public static implicit operator MyCollection2<T>(MyCollection1<T> c) => new(c._list);
+                }
+                [CollectionBuilder(typeof(MyCollectionBuilder), nameof(MyCollectionBuilder.Create2))]
+                ref struct MyCollection2<T>
+                {
+                    private readonly List<T> _list;
+                    public MyCollection2(List<T> list) { _list = list; }
+                    public IEnumerator<T> GetEnumerator() => _list.GetEnumerator();
+                }
+                static class MyCollectionBuilder
+                {
+                    public static MyCollection1<T> Create1<T>(scoped ReadOnlySpan<T> items)
+                    {
+                        return new MyCollection1<T>(new List<T>(items.ToArray()));
+                    }
+                    public static MyCollection2<T> Create2<T>(scoped ReadOnlySpan<T> items)
+                    {
+                        return new MyCollection2<T>(new List<T>(items.ToArray()));
+                    }
+                }
+                class Program
+                {
+                    static void F1<T>(MyCollection1<T> c) { Console.WriteLine("MyCollection1<T>"); }
+                    static void F1<T>(MyCollection2<T> c) { Console.WriteLine("MyCollection2<T>"); }
+                    static void F2(MyCollection2<object> c) { Console.WriteLine("MyCollection2<object>"); }
+                    static void F2(MyCollection1<object> c) { Console.WriteLine("MyCollection1<object>"); }
+                    static void Main()
+                    {
+                        F1([1, 2, 3]);
+                        F2([4, null]);
```

Comment:
I didn't understand why `MyCollection1` is better in this case and in the `F1()` call.

Reply:
`MyCollection1<T>` is a [*better conversion target*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/expressions.md#11646-better-conversion-target) because `MyCollection1<T>` is implicitly convertible to `MyCollection2<T>`.

Reply:
Discussed offline. Essentially the source and target of an implicit conversion are analogous to a derived and base type. Derived types are "better" in overload resolution.